Libraries/LibGC/ForeignCell.cpp

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2024, Andrew Kaster <andrew@ladybird.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <LibGC/DeferGC.h>
+#include <LibGC/ForeignCell.h>
+#include <LibGC/Heap.h>
+
+namespace GC {
+
+void* ForeignCell::foreign_data()
+{
+    // !!!
+    return static_cast<void*>(reinterpret_cast<u8*>(this) + sizeof(ForeignCell));
+}
+
+ForeignCell::ForeignCell(ForeignCell::Vtable vtable)
+    : m_destructor(vtable.destroy)
+    , m_finalizer(vtable.finalize)
+    , m_visit_edges(vtable.visit_edges)
+    , m_foreign_class(move(vtable.foreign_class))
+{
+    if (vtable.initialize)
+        vtable.initialize(foreign_data(), &m_foreign_class, *this);
+}
+
+ForeignCell::~ForeignCell()
+{
+    if (m_destructor)
+        m_destructor(foreign_data(), &m_foreign_class);
+}
+
+Ref<ForeignCell> ForeignCell::create(Heap& heap, size_t size, ForeignCell::Vtable vtable)
+{
+    // NOTE: GC Deferral is the caller's responsibility!
+    VERIFY(heap.is_gc_deferred());
+    auto& allocator = heap.allocator_for_size(size + sizeof(ForeignCell));
```

Comment:
Ah, thanks for the pointers there on alignment. I think the simplest approach for this PR is going to be the 'round up to alignment' idea. In the future it would be nice to extend our `AllocatorForType` concept to foreign types as well, to get per-type isolation and ensure that the same vaddr is only ever used for the same type. Perhaps that's going to come along naturally with a patch to properly cache the swift type metadata.